### PR TITLE
fix and improve job history pagination

### PR DESCRIPTION
**Pull request by wdbaruni**
This PR addresses issues with our current job history pagination and introduces a more efficient and reliable approach.

### Problem:
Our existing pagination method for job history had a few quirks:
- Returned duplicate results and sometimes out of order
- Didn't clearly indicate when we've reached the end of history
- Loaded all events into memory before filtering and sorting

### Solution:
1. Single Bucket: We now use a single BoltDB bucket for both job and execution history events. This simplifies our data structure and makes queries more straightforward. Previously we had one bucket for job events and another for all execution events combined, which didn't provide much performance value, but made maintaining a single order of events more challenging.

2. BoltDB Sequence: We're now utilizing BoltDB's built-in sequences as offsets. This gives us consistent ordering and allows for faster seeks, eliminating the need to load all events into memory.

3. Clear Endings: We now have a definitive way to know when we've reached the end of history. The pagination ends (`NextToken` is empty) when:
   a. We've read the latest available history in the store, AND
   b. The job has reached a terminal state

This work is needed to fix printing job progress in a linear way instead of grouped by execution, which has issues

### CODE REVIEW

**Review by frrist — APPROVED** [review]
Question about migrations, if none are required LGTM. Otherwise I'd like to review once implemented.

File: cmd/util/printer/printers.go [hunk 1]
```diff
@@ -0,0 +1 @@
+package printer
```
[the remark this answers is not in the document]
Comment (wdbaruni) [reply cmd/util/printer/printers.go]: remove in https://github.com/bacalhau-project/bacalhau/pull/4487/commits/2cf4e639e457d8a78ea5e00b164ca203cf5053c3

File: cmd/util/printer/constants.go [hunk 2]
```diff
@@ -0,0 +1 @@
+package printer
```
[the remark this answers is not in the document]
Comment (wdbaruni) [reply cmd/util/printer/constants.go]: removed in https://github.com/bacalhau-project/bacalhau/pull/4487/commits/2cf4e639e457d8a78ea5e00b164ca203cf5053c3

File: pkg/jobstore/boltdb/store.go [hunk 3]
```diff
@@ -27,11 +26,10 @@ import (
...
 const (
-	BucketJobs             = "jobs"
-	BucketJobExecutions    = "executions"
-	BucketJobEvaluations   = "evaluations"
-	BucketJobHistory       = "job_history"
-	BucketExecutionHistory = "execution_history"
+	BucketJobs           = "jobs"
+	BucketJobExecutions  = "executions"
+	BucketJobEvaluations = "evaluations"
+	BucketJobHistory     = "history"
```
[the remark this answers is not in the document]
Comment (wdbaruni) [reply pkg/jobstore/boltdb/store.go:32]: no need. just history events for old jobs won't be retrievable, but that has minimal impact on `job describe` output

[the remark this answers is not in the document]
Comment (frrist) [reply pkg/jobstore/boltdb/store.go:32]: I think this change has introduced https://github.com/bacalhau-project/bacalhau/pull/4512 as `job describe` does fetch job history.

**Pull request merged**